src/oc/code.c

```diff
@@ -263,6 +265,31 @@ static void frameobj_clean(Frame* f) {
 	}
 }
 
+static void frame_objauto_clean(Frame* f) { /* only on error */
+  int i;
+  Symbol* sp = f->sp;
+  Datum* stkp = f->argn + 2 + sp->u.u_proc->nauto*2; /* bottom? of stack of frame */
+  for (i = sp->u.u_proc->nobjauto; i > 0; --i) {
+    Object* ob = stkp[-2*i].obj;
+    hoc_obj_unref(ob);
+  }
+  /* clean OBJECTTMP ob of ob.meth(...) */
+  if (f->obtmp_stk_index >= 0) {
+    stkp = f->argn - 2*f->nargs;
+    if ((int)(stkp - stack) != f->obtmp_stk_index) {
+      printf("Stack index inconsistency %d != %d in error cleanup of ob of ob.meth...\n",
+        (int)(stkp - stack), f->obtmp_stk_index);
+    }
+    if (stkp[1].i == OBJECTTMP) {
+      Object* ob = f->argn[-2*f->nargs].obj;
```

Comment:
`nitpick` : I don't know the logic of the code here but in general when we have expressions like `f->argn + 2` or `sp->u.u_proc->nauto*2;` or   `Object* ob = stkp[-2*i].obj` : it will be helpful to have some description in the comment about why this is the case. This way it will help a bit for other developers.

Reply:
Added some comments.